packages/api/src/utils/logs.ts

```diff
@@ -0,0 +1,310 @@
+      ...context
+    }
+
+    // This array of errors not to send to Sentry could be configurable in the
+    // constructor if we want to keep this Logging class more generic
+    const skipForSentry = this.opts.skipForSentry
+
+    if (message instanceof Error) {
+      log = {
+        ...log,
+        stack: message.stack,
+        message: message.message
+      }
+      if (
+        this.opts.sentry != null &&
+        (skipForSentry != null) &&
+        !skipForSentry.some((cls) => message instanceof Error)
+      ) {
+        this.opts.sentry.captureException(message)
+      }
+      if (this.opts?.debug != null) {
+        console[level](`[${dt}]`, message)
+      }
+    } else {
+      log = {
+        ...log,
+        message
+      }
+      if (this.opts?.debug != null) {
+        console[level](`[${dt}]`, log.message, context)
+      }
+    }
+
+    this._add(log)
+  }
+
+  /**
+   * @param {string} message
+   * @param {any} [context]
+   */
+  debug (message: string, context: ExecutionContext) {
+    return this.log(message, 'debug', context)
+  }
+
+  /**
+   * @param {string} message
+   * @param {any} [context]
+   */
+  info (message: string, context: ExecutionContext) {
+    return this.log(message, 'info', context)
+  }
+
+  /**
+   * @param {string} message
+   * @param {any} [context]
+   */
+  warn (message: string, context: ExecutionContext) {
+    return this.log(message, 'warn', context)
+  }
+
+  /**
+   * @param {string | Error} message
+   * @param {any} [context]
+   */
+  error (message: string | Error, context: ExecutionContext) {
+    return this.log(message, 'error', context)
+  }
+
+  /**
+   * Start a timer to be logged under the given name.
+   *
+   * @param {string} name
+   * @param {string} [description]
+   */
+  time (name: string, description?: string) {
+    if (this._times.get(name) != null) {
+      return console.warn(`A timer named ${name} has already been started`)
+    }
+    this._times.set(name, {
+      name: name,
+      description: description,
+      start: Date.now()
+    })
+    this._timesOrder.push(name)
+  }
+
+  /**
+   * End the timer of the given name.
+   *
+   * @param {string} name
+   */
+  timeEnd (name: string) {
+    const timeObj = this._times.get(name)
+    if (timeObj == null) {
+      return console.warn(`No such name ${name}`)
+    }
+    this._times.delete(name)
+    const end = Date.now()
+    const duration = end - timeObj.start
+    const value = duration
+    timeObj.value = value
+    this._times.set(name, {
+      ...timeObj,
+      end,
+      duration
+    })
+
+    if (this.opts?.debug != null) {
+      console.log(`[${this._date()}]`, `${name}: ${duration} ms`)
+    }
+    return timeObj
+  }
+
+  _timersString () {
+    const result = []
+    for (const key of this._timesOrder) {
+      const { name, duration, description }: { name: string, duration: string, description: string } = this._times.get(key)
+      result.push(
+        description === ''
```

Comment:
This comparison is the wrong way round, if I'm not mistaken?
```suggestion
        description !== ''
```